dev-util/jenkins-bin/files/jenkins-bin.service

```diff
@@ -2,8 +2,9 @@
 Description=Jenkins Daemon
 
 [Service]
+EnvironmentFile=/etc/jenkins/jenkins.conf
 Type=forking
-ExecStart=/usr/bin/java -Djava.awt.headless=true -DJENKINS_HOME=/var/lib/jenkins/home -jar /opt/jenkins/jenkins.war --daemon --logfile=/var/log/jenkins/jenkins.log --httpPort=8080 --debug=5 --handlerCountMax=100 --handlerCountMaxIdle=20 --accessLoggerClassName=winstone.accesslog.SimpleAccessLogger --simpleAccessLogger.format=combined --simpleAccessLogger.file=/var/log/jenkins/access_log
+ExecStart=/bin/sh -c 'eval $JENKINS_COMMAND_LINE'
```

Comment:
something like this would be better (don't use this exactly as it's just an example). not sure if forking is needed if `--daemon` is not called (I would not use `--daemon` as systemd will log itself, which could be useful).

```bash
EnvironmentFile=/etc/jenkins/jenkins.conf
ExecStart=/usr/bin/java ${JENKINS_JAVA_OPTIONS} -jar /usr/lib/jenkins/jenkins.war --httpPort=${JENKINS_PORT} --httpListenAddress=${JENKINS_LISTEN_ADDRESS} ${JENKINS_ARGS}
```

Reply:
@prometheanfire it doesn't work as "--argument", that's why you generally need a eval

Reply:
The EnvironmentFile needs to have the following.

```
AWT_HEADLESS=true
JENKINS_HOME=/var/lib/jenkins/home
JARFILE=/opt/jenkins/jenkins.war
LOGFILE=/var/log/jenkins/jenkins.log
HTTP_PORT=8080
DEBUG_LEVEL=5
HANDLER_COUNT_MAX=100
HANDLER_COUNT_IDLE=20
LOGGER_CLASS=winstone.accesslog.SimpleAccessLogger
LOGGER_FORMAT=combined
ACCESS_LOG_FILE=/var/log/jenkins/access_log
```


and the unit file should be the following, I removed the daemon and set type to simeple to let systemd handle it (otherwise a pidfile may be useful).

```bash
[Unit]
 Description=Jenkins Daemon

[Service]
EnvironmentFile=/etc/jenkins/jenkins.conf
Type=simple
ExecStart=/usr/bin/java -Djava.awt.headless=${AWT_HEADLESS} -DJENKINS_HOME=${JENKINS_HOME} -jar ${JARFILE} --logfile=${LOGFILE --httpPort=${HTTP_PORT} --debug=${DEBUG_LEVEL} --handlerCountMax=${HANDLER_COUNT_MAX} --handlerCountMaxIdle=${HANDLER_COUNT_IDLE} --accessLoggerClassName=${LOGGER_CLASS} --simpleAccessLogger.format=${LOGGER_FORMAT} --simpleAccessLogger.file=${ACCESS_LOG_FILE}
```

add other options as needed

Reply:
As per discussion, @graaff / @prometheanfire , I removed the /bin/sh -c 'eval' method, it can go definitely wrong and that's only a workaround, not something we could definitely use on the longterm plan

Added sane individiual variables that can be used from /etc/jenkins/jenkins.conf, some of them are hardcoded by default for functionality reasons ( check Georg Voigtländer proposal from bugs.gentoo.org/665030 )
I expanded Georg's config and added additional values which will be used in the ExecStart=(..), and changed from /etc/jenkins/jenkins.conf

The conditional usage of 'systemd' I would suggest to not change due to usage of FILESDIR helpers systemd service file and not upstream-maintained